flaml/integrations/oai/completion.py

```diff
@@ -172,25 +195,41 @@ def eval(cls, config: dict, prune=True, eval_only=False):
         Args:
             config (dict): Hyperparameter setting for the openai api call.
             prune (bool, optional): Whether to enable pruning. Defaults to True.
-            eval_only (bool, optional): Whether to evaluate only. Defaults to False.
+            eval_only (bool, optional): Whether to evaluate only (ignore the inference budget and no timeout).
+              Defaults to False.
 
         Returns:
             dict: Evaluation results.
         """
         cost = 0
         data = cls.data
+        model = config["model"]
         data_length = len(data)
-        target_n_tokens = (
-            1000 * cls.inference_budget / cls.price1K[config["model"]]
-            if cls.inference_budget and cls.price1K.get(config["model"])
+        target_n_tokens = getattr(cls, "inference_budget", None) and (
+            1000 * cls.inference_budget / cls.price1K[model]
+            if cls.inference_budget and cls.price1K.get(model)
             else None
         )
-        prune_hp = cls._prune_hp
+        prune_hp = getattr(cls, "_prune_hp", "n")
         metric = cls._metric
-        config_n = config[prune_hp]
+        config_n = config.get(prune_hp, 1)  # default value in OpenAI is 1
         max_tokens = config.get("max_tokens", 16)  # default value in OpenAI is 16
         region_key = cls._get_region_key(config)
-        prompt = cls._prompts[config["prompt"]]
+        if model in cls.chat_models:
+            # either "prompt" should be in config (for being compatible with non-chat models)
+            # or "messages" should be in config (for tuning chat models only)
+            prompt = config.get("prompt")
+            if prompt is None:
+                messages = config.get("messages")
+                if messages is None:
+                    raise ValueError(
+                        "Either prompt or messages should be in config for chat models."
+                    )
+                messages = cls._messages[messages]
+            else:
+                prompt = cls._prompts[prompt]
```

Comment:
should check the case both prompt and messages are non empty?

Reply:
> [Chat](https://github.com/microsoft/FLAML/blob/3e568ca20b44402caa60ee76bfffdbc47d54ab75/notebook/integrate_chatgpt_code.ipynb) and [Completion](https://github.com/microsoft/FLAML/blob/3e568ca20b44402caa60ee76bfffdbc47d54ab75/notebook/integrate_chatgpt_code.ipynb) have the same HELM success rate (46%) on HumanEval? Possible typo?

The 46% is the number from HELM. HELM doesn't have numbers for Chat.

Reply:
addressed